### PR TITLE
Cap execution of docker image fetch

### PR DESCRIPTION
**Pull request by kamaradclimber**
When docker image fetching is stalled, mesos correctly timeout (thanks
to executor_registration_timeout) however container destroy simply waits
for the image provisioning to finish. This leads to task being "staged"
forover (until mesos agent restart or curl being killed).

While it might be difficult to timeout on mesos side, we can simply
leverage curl timeout to have provisioning being eventually interrupted.

This is not fully robust and poorly configurable (I've leveraged the
stall timeout option instead of of the executor_registration_timeout to
avoid passing references to this option in all places). But it should be
sufficient.

Change-Id: Ic301450b71a82f13ec15bf40d603bf754ae6fc9a
Reference: https://confluence.criteois.com/x/jakyJQ

### CODE REVIEW

File: src/uri/fetchers/docker.cpp [hunk 1]
```diff
@@ -268,6 +276,13 @@ static Future<int> download(
   argv.push_back("--connect-timeout");
   argv.push_back("30");
 
+  // to avoid introducing a new option, we simply reuse stall_timeout option
+  // for container image layers we simply multiply this value by a constant factor
+  if (stallTimeout.isSome()) {
+    argv.push_back("--max-time");
+    argv.push_back(std::to_string(5 * static_cast<long>(stallTimeout->secs())));
```
Comment (pierresouchay) [inline src/uri/fetchers/docker.cpp:286]: 5 sec to fetch a docker image, sounds a bit small to me no?
Just DNS might (with retry) reach quickly 5s...

You might maybe use `--connect-timeout 5` and --max-time something_bigger, what do you think?

Comment (kamaradclimber) [reply src/uri/fetchers/docker.cpp:286]: stallTimeout value is 60s by default, to it gives 5minutes per layer (which is higher than the executor_registration_timeout)

Comment (pierresouchay) [reply src/uri/fetchers/docker.cpp:286]: huh, ok, sorry, make sense, (I did not see stallTimeout value and wrongly interpreted it as 5s)

**Comment by kamaradclimber** [comment]
@Mergifyio refresh

**Review by Lqp1 — DISMISSED** [review]
LGTM, but if you have some time I would gladly see more context in the comment block you added:
* why the assumption of using 5*stallTimeout is sufficient
* why we are adding that - just one line, because commit messages are not eternal

**Comment by kamaradclimber** [comment]
> Commit messages are not eternal

I'd expect commit messages to be preserved, this is the really the main reason why we write them.

About the 5 seconds, it is purely empirical:
- stall timeout is 60s (on our infra at least)
- 5 minutes is the registration_timeout

**Comment by kamaradclimber** [comment]
updated (we just lost 2h of pre-submit :( )

**Comment by Lqp1** [comment]
If it was passing when I wrote you can just push this commit on the destination branch as you only edited a comment, results won't change.

**Review by fquesnel — APPROVED** [review]
Lgtm

**Pull request merged**